cmd/openshift-install/migrate.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"github.com/spf13/cobra"
+
+	azmigrate "github.com/openshift/installer/pkg/migrate/azure"
+)
+
+var (
+	azureMigrateEligibleOpts struct {
+		public bool
+	}
+)
+
+func runMigrateAzureDNSEligibleCmd(cmd *cobra.Command, args []string) error {
+	return azmigrate.DoEligible(azureMigrateEligibleOpts.public)
+}
+
+func newMigrateAzureDNSEligibleCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "eligible",
+		Short: "Show legacy zones that are eligible to be migrated",
+		RunE:  runMigrateAzureDNSEligibleCmd,
+	}
+
+	cmd.PersistentFlags().BoolVar(&azureMigrateEligibleOpts.public, "public", false, "Show legacy public zones")
+
+	return cmd
+}
+
+var (
+	azureMigrateListOpts struct {
+		legacyZone    string
+		privateZone   string
+		resourceGroup string
+		all           bool
+		public        bool
+	}
+)
+
+func runMigrateAzureDNSListCmd(cmd *cobra.Command, args []string) error {
+	if azureMigrateListOpts.all == true {
+		return azmigrate.DoListAllZones(azureMigrateListOpts.public)
+	}
+
+	if (azureMigrateListOpts.legacyZone == "" && azureMigrateListOpts.privateZone == "") || azureMigrateListOpts.resourceGroup == "" {
+		return cmd.Help()
+	}
+
+	if azureMigrateListOpts.legacyZone != "" {
+		return azmigrate.DoListLegacyZone(azureMigrateListOpts.resourceGroup, azureMigrateListOpts.legacyZone, azureMigrateListOpts.public)
+	}
+
+	return azmigrate.DoListPrivateZone(azureMigrateListOpts.resourceGroup, azureMigrateListOpts.privateZone)
+}
+
+func newMigrateAzureDNSListCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
```

Comment:
why is this required, can't users just use az to list a zone?

Reply:
I've removed the leftover cruft.